dmoj/cptbox/_cptbox.pyx

```diff
@@ -417,25 +423,25 @@ cdef class Process:
         else:
             config.max_fd = 2 + len(fds)
             config.fds = <int*>malloc(sizeof(int) * len(fds))
-            for i in xrange(len(fds)):
+            for i in range(len(fds)):
                 config.fds[i] = fds[i]
 
-        config.debugger_type = self.debugger._debugger_type
-        config.trace_syscalls = self._trace_syscalls
-        config.syscall_whitelist = <bint*>malloc(sizeof(bint) * MAX_SYSCALL_NUMBER)
-        for i in xrange(MAX_SYSCALL_NUMBER):
-            # We have to force exit syscalls to trap, so that we can be notified
-            # that the process spawned successfully when using `seccomp`. Otherwise,
-            # a simple assembly program could terminate without ever trapping.
-            if not self.debugger.is_exit(i):
-                config.syscall_whitelist[i] = self._syscall_whitelist[i]
-            else:
-                config.syscall_whitelist[i] = False
+        if self.use_seccomp:
+            config.avoid_seccomp = False
+            config.abi_for_seccomp, whitelist = self._get_seccomp_abi_whitelist()
+            config.seccomp_whitelist = <bint*>malloc(sizeof(bint) * MAX_SYSCALL_NUMBER)
+            for i in range(MAX_SYSCALL_NUMBER):
+                config.seccomp_whitelist[i] = i < len(whitelist) and whitelist[i]
```

Comment:
`i < len(whitelist)` can't ever be true, can it? I feel we probably want to assert `len(whitelist) == MAX_SYSCALL_NUMBER`, if anything